### PR TITLE
refactor: check for global stack traces instead of forcing NewGlobal

### PR DESCRIPTION
**Pull request by morningvera**

### CODE REVIEW

File: stack.go [hunk 1]
```diff
@@ -127,6 +127,17 @@ func (s *stack) get() []StackFrame {
 	return sFrames
 }
 
+// isGlobal determines if the stack trace represents a global error
+func (s *stack) isGlobal() bool {
+	frames := s.get()
+	for _, f := range frames {
+		if f.Name == "runtime.doInit" {
```
Comment (sum2000) [inline stack.go]: should we compare lowercase of each? i dont think it is necessary but mot sure if f.Name will ever be upper or camelcase

File: examples_test.go [hunk 2]
```diff
@@ -191,16 +192,12 @@ func TestExampleToString_external(t *testing.T) {
 	ExampleToString_external()
 }
 
-// Demonstrates string formatting of wrapped errors that originate from global root errors (created
-// via eris.NewGlobal). You can try this example in the Go playground
-// (https://play.golang.org/p/8YgyDwk9xBJ).
+// Demonstrates string formatting of wrapped errors that originate from global root errors. You can
+// try this example in the Go playground (https://play.golang.org/p/8YgyDwk9xBJ).
```
Comment (sum2000) [inline examples_test.go]: are you planning to update the example links later?

Comment (morningvera) [reply examples_test.go]: yeah, can't update them until after release

File: stack.go (hunk 1 above)
Comment (morningvera) [reply stack.go]: it's safer and easy enough to change so i'll do it

**Comment by sum2000** [comment]
Until we don't release a new version, people will still be using NewGlobal so maybe we should not updated docs and example until then?

**Pull request merged**